src/services/review/__tests__/ReviewRequestService.spec.ts

```diff
@@ -0,0 +1,1151 @@
+        reviewId: MOCK_REVIEW_REQUEST_ONE.id,
+        pullRequestNumber: mockPullRequestNumber,
+        reviewLink: `cms.isomer.gov.sg/sites/${mockUserWithSiteSessionData.siteName}/review/${mockPullRequestNumber}`,
+      })
+    })
+  })
+
+  describe("listReviewRequest", () => {
+    // NOTE: We are only assuming one review request is returned
+    it("should return an array of basic review request objects not viewed before", async () => {
+      // Arrange
+      const expected = [
+        {
+          id: MOCK_REVIEW_REQUEST_ONE.id,
+          author: MOCK_IDENTITY_EMAIL_ONE,
+          status: MOCK_REVIEW_REQUEST_ONE.reviewStatus,
+          title: MOCK_PULL_REQUEST_ONE.title,
+          description: MOCK_PULL_REQUEST_ONE.body,
+          changedFiles: MOCK_PULL_REQUEST_ONE.changed_files,
+          createdAt: new Date(MOCK_PULL_REQUEST_ONE.created_at).getTime(),
+          newComments: 2,
+          firstView: true,
+        },
+      ]
+      MockReviewRequestRepository.findAll.mockResolvedValueOnce([
+        MOCK_REVIEW_REQUEST_ONE,
+      ])
+      MockReviewApi.getPullRequest.mockResolvedValueOnce(MOCK_PULL_REQUEST_ONE)
+      MockReviewRequestViewRepository.count.mockResolvedValueOnce(0)
+      MockReviewApi.getComments.mockResolvedValueOnce([
+        MOCK_GITHUB_COMMENT_DATA_ONE,
+        MOCK_GITHUB_COMMENT_DATA_TWO,
+      ])
+      MockReviewRequestViewRepository.findOne.mockResolvedValueOnce(null)
+      MockUsersRepository.findByPk.mockResolvedValueOnce({
+        email: MOCK_IDENTITY_EMAIL_ONE,
+      })
+      MockUsersRepository.findByPk.mockResolvedValueOnce({
+        email: MOCK_IDENTITY_EMAIL_TWO,
+      })
+
+      // Act
+      const actual = await ReviewRequestService.listReviewRequest(
+        mockUserWithSiteSessionData,
+        mockSiteOrmResponseWithAllCollaborators as Attributes<Site>
+      )
+
+      // Assert
+      expect(actual).toEqual(expected)
+      expect(MockReviewRequestRepository.findAll).toHaveBeenCalled()
+      expect(MockReviewApi.getPullRequest).toHaveBeenCalledWith(
+        mockUserWithSiteSessionData.siteName,
+        MOCK_REVIEW_REQUEST_ONE.reviewMeta.pullRequestNumber
+      )
+      expect(MockReviewRequestViewRepository.count).toHaveBeenCalled()
+      expect(SpyReviewRequestService.getComments).toHaveBeenCalledWith(
+        mockUserWithSiteSessionData,
+        mockSiteOrmResponseWithAllCollaborators as Attributes<Site>,
+        MOCK_REVIEW_REQUEST_ONE.reviewMeta.pullRequestNumber
+      )
+      expect(MockReviewApi.getComments).toHaveBeenCalledWith(
+        mockUserWithSiteSessionData.siteName,
+        MOCK_REVIEW_REQUEST_ONE.reviewMeta.pullRequestNumber
+      )
+      expect(SpyReviewRequestService.computeCommentData).toHaveBeenCalled()
+      expect(MockUsersRepository.findByPk).toHaveBeenCalledTimes(2)
+    })
+
+    it("should return an array of basic review request objects with one new comment", async () => {
```

Comment:
could i ask why the `one new comment` is required here

Reply:
Adjusted in a705ca2.